src/modules/avformat/consumer_avformat.c

```diff
@@ -1766,36 +1768,16 @@ static void *consumer_thread( void *arg )
 
 					if ( mlt_properties_get_int( frame_properties, "rendered" ) )
 					{
-						int i = 0;
-						uint8_t *p;
-						uint8_t *q;
-						int stride = mlt_image_format_size( img_fmt, width, 0, NULL );
-
+						AVFrame video_avframe;
 						mlt_frame_get_image( frame, &image, &img_fmt, &img_width, &img_height, 0 );
-						q = image;
 
-						// Convert the mlt frame to an AVPicture
-						if ( img_fmt == mlt_image_yuv420p )
-						{
-							stride = width * height;
-							memcpy( video_avframe->data[0], q, video_avframe->linesize[0] * height );
-							q += stride;
-							memcpy( video_avframe->data[1], q, video_avframe->linesize[1] * height / 2 );
-							q += stride / 4;
-							memcpy( video_avframe->data[2], q, video_avframe->linesize[2] * height / 2 );
-						}
-						else for ( i = 0; i < height; i ++ )
-						{
-							p = video_avframe->data[0] + i * video_avframe->linesize[0];
-							memcpy( p, q, stride );
-							q += stride;
-						}
+						mlt_image_format_planes( img_fmt, width, height, image, video_avframe.data, video_avframe.linesize );
```

Comment:
@ddennedy have you objections against this? new code avoiding one more memory copying...